lib/src/main/java/com/orange/ods/compose/component/appbar/top/OdsLargeTopAppBar.kt

```diff
@@ -0,0 +1,163 @@
+import androidx.compose.material3.LargeTopAppBar
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.material3.TopAppBarScrollBehavior
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import com.orange.ods.R
+import com.orange.ods.compose.component.OdsComposable
+import com.orange.ods.compose.component.menu.OdsDropdownMenuItem
+import com.orange.ods.compose.component.utilities.BasicPreviewParameterProvider
+import com.orange.ods.compose.component.utilities.Preview
+import com.orange.ods.compose.component.utilities.UiModePreviews
+import com.orange.ods.compose.theme.OdsTheme
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+@OdsComposable
+fun OdsLargeTopAppBar(
+    title: String,
+    modifier: Modifier = Modifier,
+    navigationIcon: @Composable () -> Unit = {},
+    onNavigationIconClick: () -> Unit = {},
+    actions: @Composable RowScope.() -> Unit = {},
+    scrollBehavior: TopAppBarScrollBehavior? = null
+) {
+    val contentColor = OdsTheme.colors.component.topAppBar.barContent
+    val expandedTitleStartPadding = 48.dp
+    val collapsedTitleStartPadding = 8.dp
+    val expandedTitleAlpha = 1f
+    val expandedTitleMaxLines = 2
+    val collapsedTitleMaxLines = 1
+    val stateChangeFraction = 0.7
+
+    val titleStartPadding by remember {
+        derivedStateOf {
+            if (scrollBehavior != null && scrollBehavior.state.collapsedFraction >= 0.85) collapsedTitleStartPadding else expandedTitleStartPadding
+        }
+    }
+
+    val titleAlpha by remember {
+        derivedStateOf {
+            if (scrollBehavior != null) {
+                when (scrollBehavior.state.collapsedFraction) {
+                    in 0.0..stateChangeFraction -> 1 - (scrollBehavior.state.collapsedFraction * 1.5)
+                    in (stateChangeFraction + 0.15)..1.0 -> 0 + scrollBehavior.state.collapsedFraction
+                    else -> 0
+                }.toFloat()
+            } else {
+                expandedTitleAlpha
+            }
+        }
+    }
+    val titleMaxLines by remember {
+        derivedStateOf {
+            if (scrollBehavior != null && scrollBehavior.state.collapsedFraction >= stateChangeFraction) collapsedTitleMaxLines else expandedTitleMaxLines
+        }
+    }
+
+    LargeTopAppBar(
+        title = {
+            Text(
+                modifier = Modifier
+                    .padding(
+                        start = if (scrollBehavior != null) titleStartPadding else expandedTitleStartPadding,
```

Comment:
This is the question I had. Try to do that, set the scroll behavior to `None` in the customization bottom sheet and scroll the content... The scroll is handled even if the scroll behavior is null. Maybe there is another fix to avoid this...

Reply:
I found a fix. We need to use `scrollBehavior` as a key when calling `remember`, otherwise state is not properly updated:

```kotlin
val titleStartPadding by remember(scrollBehavior) {
```

```kotlin
val titleAlpha by remember(scrollBehavior) {
```

```kotlin
val titleMaxLines by remember(scrollBehavior) {
```